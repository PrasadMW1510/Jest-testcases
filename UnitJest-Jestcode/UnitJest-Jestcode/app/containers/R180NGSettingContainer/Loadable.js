/**
 *
 * Asynchronously loads the component for R180NGSetting
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
