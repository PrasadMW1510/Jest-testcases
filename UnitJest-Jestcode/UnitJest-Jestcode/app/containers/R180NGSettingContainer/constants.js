/*
 *
 * R180NGSettingContainer constants
 *
 */

export const R180NG_PROGRAM_SETTINGS_REQUEST = 'R180NG_program_settings_request';

export const R180NG_PROGRAM_SETTINGS_REQUEST_SUCCESS = 'R180NG_program_settings_request_success';

export const R180NG_PROGRAM_SETTINGS_REQUEST_FAILURE = 'R180NG_program_settings_request_failure';

export const R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST =
  'R180NG_program_settings_enrollment_request';

export const R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_SUCCESS =
  'R180NG_program_settings_enrollment_request_success';

export const R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_FAILURE =
  'R180NG_program_settings_enrollment_request_failure';

export const R180NG_SAVE_REQUEST = 'r180ng_save_request';

export const R180NG_SAVE_REQUEST_SUCCESS = 'r180ng_save_request_success';

export const R180NG_SAVE_REQUEST_FAILURE = 'r180ng_save_request_failure';

export const UPDATE_R180NG_SETTING_REQUEST_SUCCESS = 'update_r180ngsetting_request_success';

export const R180NG_ENROLLMENT_PROGRAMS = [
  'READ 180 NG Stage A',
  'READ 180 NG Stage B',
  'READ 180 NG Stage C',
];
