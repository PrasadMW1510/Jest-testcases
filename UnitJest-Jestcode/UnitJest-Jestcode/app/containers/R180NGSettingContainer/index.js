/**
 *
 * R180NgsettingContainer
 *
 */

export { default } from './R180NGSettingContainer';
