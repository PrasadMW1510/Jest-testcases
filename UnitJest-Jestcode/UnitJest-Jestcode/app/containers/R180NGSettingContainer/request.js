import API, { getBaseUrlWithoutSlmsObject } from 'utils/request';
/**
 * Gets Program Group settings for R180NG
 *
 * @param sessionId
 * @param cohortId
 */

export const getGroupSettingsR180NG = (sessionId, cohortId, cohortType) =>
  getBaseUrlWithoutSlmsObject
    .get('/r180ng/r180ngProductCtrls', {
      params: {
        command: 'get_group_settings',
        sid: sessionId,
        cohort_id: cohortId,
        cohort_type: cohortType,
      },
    })
    .then(response => response.output.output_data[0].group_settings[0]);

/**
 * Gets Program Grade settings for R180NG
 *
 * @param sessionId
 * @param cohortId
 */

export const getGradeSettingsR180NG = (sessionId, cohortId, cohortType, schoolId) =>
  getBaseUrlWithoutSlmsObject
    .get('/r180ng/r180ngProductCtrls', {
      params: {
        command: 'get_group_settings',
        sid: sessionId,
        cohort_id: cohortId,
        cohort_type: cohortType,
        school_id: schoolId,
      },
    })
    .then(response => response.output.output_data[0].group_settings[0]);

/**
 * Gets Program Enrollment Count of a user District
 *
 * @param sessionId
 * @param districtId
 */
export const getEnrollmentCountDistrict = (sessionId, districtId) =>
  API.get('/SlmsDistrict', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      district_id: districtId,
    },
  }).then(response => response.output_data[0].applications[0].application);

/**
 * Gets Program Enrollment Count of a user District
 *
 * @param sessionId
 * @param schoolId
 */
export const getEnrollmentCountSchool = (sessionId, schoolId) =>
  API.get('/SlmsSchool', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      school_id: schoolId,
    },
  }).then(response => response.output_data[0].applications[0].application);

/**
 * Gets Program Enrollment Count of a user Teacher
 *
 * @param sessionId
 * @param userId
 */
export const getEnrollmentCountTeacher = (sessionId, userId) =>
  API.get('/SlmsTeacher', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      user_id: userId,
    },
  }).then(response => response.output_data[0].applications[0].application);

/**
 * Gets Program Enrollment Count of a user Class
 *
 * @param sessionId
 * @param userId
 */
export const getEnrollmentCountClass = (sessionId, classId) =>
  API.get('/SlmsClass', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      class_id: classId,
    },
  }).then(response => response.output_data[0].applications[0].application);
/**
 * Gets Program Enrollment Count of a Grade
 *
 * @param sessionId
 * @param userId
 */
export const getEnrollmentCountGrade = (sessionId, schoolId, gradeNo) =>
  API.get('/SlmsGrade', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      school_id: schoolId,
      grade: gradeNo,
    },
  }).then(response => response.output_data[0].applications[0].application);

/**
 * Gets Program Enrollment Count of a Group
 *
 * @param sessionId
 * @param userId
 */
export const getEnrollmentCountGroup = (sessionId, groupId) =>
  API.get('/SlmsGroup', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      group_id: groupId,
    },
  }).then(response => response.output_data[0].applications[0].application);
/**
 * Gets Program Enrollment Count of a user Student
 *
 * @param sessionId
 * @param userId
 */
export const getEnrollmentCountStudent = (sessionId, userId) =>
  API.get('SlmsStudent', {
    params: {
      command: 'get_enrollment_count',
      sid: sessionId,
      user_id: userId,
    },
  }).then(response => response.output_data[0].applications[0].application);
/**
 *Changes Settings form Program Settings for R180NG
 *
 * @param sessionId
 * @param changesXML
 * @returns {PromiseLike<{login: {}[]}> | Promise<{login: {}[]}> | *}
 * r180ng/r180ngProductCtrls?sid=0uov9cg32e9jf4hlg4177bav_2efa7f0&command=set_group_settings
 */
export const postChangeSettingsR180NG = (sessionId, changesXML) =>
  getBaseUrlWithoutSlmsObject
    .post('/r180ng/r180ngProductCtrls', changesXML, {
      params: {
        command: 'set_group_settings',
        sid: sessionId,
      },
    })
    .then(response => response.output_data);
