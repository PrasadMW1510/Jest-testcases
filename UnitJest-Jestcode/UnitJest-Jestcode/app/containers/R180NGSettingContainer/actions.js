/*
 *
 * R180NGSettingContainer actions
 *
 */

import * as Constants from './constants';

/**
 * Get the Program Settings
 * @param programSetting
 * @returns {{type}}
 */
export function R180NGProgramSettingsRequest() {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_REQUEST,
  };
}

/**
 * Dispatched when we get the Program Settings successfully
 * @param programSetting
 * @returns {{type, programSetting: *}}
 */
export function R180NGProgramSettingsRequestSuccess(programSetting = {}) {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_REQUEST_SUCCESS,
    programSetting,
  };
}

/**
 * Dispatched when the Program Settings request fails
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function R180NGProgramSettingsRequestFailure(error) {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_REQUEST_FAILURE,
    error,
  };
}

/**
 * Get the Program Settings summary
 *
 * @returns {{type}}
 */
export function R180NGProgramSettingsEnrollmentRequest() {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST,
  };
}

/**
 * Dispatched when we get the Program Settings successfully
 * @param programEnrollmentSetting
 * @returns {[type, programEnrollmentSetting: *]}
 */
export function R180NGProgramSettingsEnrollmentRequestSuccess(programEnrollmentSetting = []) {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_SUCCESS,
    programEnrollmentSetting,
  };
}
/**
 * Dispatched when the Program Settings request fails
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function R180NGProgramSettingsEnrollmentRequestFailure(error) {
  return {
    type: Constants.R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_FAILURE,
    error,
  };
}

/**
 * Handles API Logic for Changing  Program Settings for R180NG
 *
 * @param programsSettingChanged
 * @returns {{type, messageIdsChecked: *}}
 */
export function R180NGSaveRequest(programsSettingChanged = {}) {
  return {
    type: Constants.R180NG_SAVE_REQUEST,
    programsSettingChanged,
  };
}

/**
 * Handles API logic for when Changing  Program Settings for R180NG are successful
 *
 * @returns {{type}}
 */
export function R180NGSaveRequestSuccess() {
  return {
    type: Constants.R180NG_SAVE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when Changing  Program Settings for R180NG are a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function R180NGSaveRequestFailure(error) {
  return {
    type: Constants.R180NG_SAVE_REQUEST_FAILURE,
    error,
  };
}

/**
 * Handles API logic for when Changing  Program Settings for R180NG are a failure
 * @param setting
 * @returns {{type, setting: {}}}
 */
export function updateR180NGSettingRequestSuccess(setting = {}) {
  return {
    type: Constants.UPDATE_R180NG_SETTING_REQUEST_SUCCESS,
    setting,
  };
}
