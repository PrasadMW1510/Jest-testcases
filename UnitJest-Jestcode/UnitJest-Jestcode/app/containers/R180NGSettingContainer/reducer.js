/*
 *
 * R180NgsettingContainer reducer
 *
 */

import { fromJS } from 'immutable';
import { LOCATION_CHANGE } from 'react-router-redux';
import * as SmartBarConstants from 'containers/SmartBarContainer/constants';
import * as Constants from './constants';

const initialState = fromJS({
  error: false,
  loading: false,
  programSetting: {},
  programEnrollmentSetting: [],
});

function r180NGSettingContainerReducer(state = initialState, action) {
  switch (action.type) {
    case SmartBarConstants.SCHOOL_SELECTION_SUCCESS:
    case SmartBarConstants.GRADE_SELECTION_SUCCESS:
    case SmartBarConstants.TEACHER_SELECTION_SUCCESS:
    case SmartBarConstants.CLASS_SELECTION_SUCCESS:
    case SmartBarConstants.GROUP_SELECTION_SUCCESS:
    case SmartBarConstants.STUDENT_SELECTION_SUCCESS:
      return initialState.set('loading', true);
    case Constants.R180NG_PROGRAM_SETTINGS_REQUEST_SUCCESS:
      return state.set('programSetting', fromJS(action.programSetting)).set('loading', false);
    case Constants.R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_SUCCESS:
      return state.set('programEnrollmentSetting', fromJS(action.programEnrollmentSetting));
    case Constants.R180NG_PROGRAM_SETTINGS_ENROLLMENT_REQUEST_FAILURE:
    case Constants.R180NG_PROGRAM_SETTINGS_REQUEST_FAILURE:
      return state.set('error', action.error);
    case Constants.R180NG_SAVE_REQUEST_SUCCESS:
      return state.set('loading', true);
    case Constants.UPDATE_R180NG_SETTING_REQUEST_SUCCESS:
      return state.set('programSetting', fromJS(action.setting)).set('loading', false);
    case LOCATION_CHANGE:
      return action.payload.pathname === '/roster' ? initialState : state;
    case Constants.R180NG_SAVE_REQUEST_FAILURE:
      return state.set('error', action.error);
    default:
      return state;
  }
}

export default r180NGSettingContainerReducer;
