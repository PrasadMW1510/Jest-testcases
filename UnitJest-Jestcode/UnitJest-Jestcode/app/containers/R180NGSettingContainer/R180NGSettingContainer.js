/**
 *
 * R180NgsettingContainer
 *
 */

import PropTypes from 'prop-types';
import { fromJS } from 'immutable';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import R180NGSetting from 'components/R180NGSetting';
import { makeSelectLoginData, makeSelectUserOrgUserType } from 'containers/App/selectors';
import { showProgramSettingSetStudentLevelModal } from 'containers/ModalController/actions';
import {
  makeSelectSmartBarContainer,
  selectSmartBarContainerDomain,
} from 'containers/SmartBarContainer/selectors';
import makeSelectRosterPage from 'containers/RosterPage/selectors';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';

import {
  R180NGProgramSettingsRequest,
  R180NGProgramSettingsEnrollmentRequest,
  R180NGSaveRequest,
} from './actions';
import { R180NG_ENROLLMENT_PROGRAMS } from './constants';
import reducer from './reducer';
import saga from './saga';
import { makeProgramSetting, makeEnrollmentSetting, selectProgramSetting } from './selectors';

export class R180NGSettingContainer extends React.Component {
  componentDidMount() {
    this.props.R180NGProgramSettingsRequest();
    this.props.R180NGProgramSettingsEnrollmentRequest();
  }

  shouldComponentUpdate(nextProps) {
    return (
      this.props.programSettingData.get('programSetting') !==
        nextProps.programSettingData.get('programSetting') ||
      this.props.programSettingData.get('programEnrollmentSetting') !==
        nextProps.programSettingData.get('programEnrollmentSetting')
    );
  }

  getCohortType() {
    return this.props.smartBarSelections.getIn(['selectedCohType']);
  }

  getUserType() {
    return this.props.userType;
  }

  getEnroll() {
    return this.props.programEnrollmentCount && this.props.programEnrollmentCount.toJS();
  }

  getEnrollmentCount() {
    let items = this.getEnroll();
    items = items.programEnrollmentSetting;
    if (typeof items === 'undefined' || items.length === 0 || items === 'admin') {
      return 0;
    }

    const programEnrolledCount = items.filter(item =>
      Object.keys(item).some(
        k =>
          item[k] != null &&
          (item[k].toString().includes(R180NG_ENROLLMENT_PROGRAMS[0]) ||
            item[k].toString().includes(R180NG_ENROLLMENT_PROGRAMS[1]) ||
            item[k].toString().includes(R180NG_ENROLLMENT_PROGRAMS[2]))
      )
    );
    const total = programEnrolledCount.reduce(
      (prevVal, elem) => prevVal + Number(elem.students[0].total[0]),
      0
    );
    return total;
  }

  handleSaveClick = programsSettingChanged => {
    this.props.R180NGSaveRequest(programsSettingChanged);
  };

  handleSetStudentLevelClick = e => {
    this.props.showProgramSettingSetStudentLevelModal(e);
  };

  render() {
    return (
      <div>
        <R180NGSetting
          enrollmentData={this.props.enrollmentCount}
          programName={this.props.rosterPage.selectedProgram}
          programSettingData={this.props.programSettingData && this.props.programSettingData.toJS()}
          onStudentLevelClick={this.handleSetStudentLevelClick}
          handleSaveClick={this.handleSaveClick}
          smartBarSelections={this.getCohortType()}
          userType={this.getUserType()}
          isLoading={this.props.programSettingData.get('loading')}
        />
      </div>
    );
  }
}
R180NGSettingContainer.defaultProps = {
  enrollmentCount: 0,
  programSettingData: fromJS({
    loading: false,
    programSetting: {},
  }),
};

R180NGSettingContainer.propTypes = {
  showProgramSettingSetStudentLevelModal: PropTypes.func.isRequired,
  rosterPage: PropTypes.any,
  R180NGProgramSettingsRequest: PropTypes.any.isRequired,
  R180NGProgramSettingsEnrollmentRequest: PropTypes.any.isRequired,
  programSettingData: PropTypes.object.isRequired,
  programEnrollmentCount: PropTypes.any,
  R180NGSaveRequest: PropTypes.func,
  smartBarSelections: PropTypes.any,
  enrollmentCount: PropTypes.number,
  userType: PropTypes.any,
};

const mapStateToProps = createStructuredSelector({
  loginData: makeSelectLoginData(),
  rosterPage: makeSelectRosterPage(),
  userType: makeSelectUserOrgUserType(),
  programSettingData: makeProgramSetting(),
  programEnrollmentCount: makeEnrollmentSetting(),
  smartBarSelections: makeSelectSmartBarContainer(),
});

const withConnect = connect(mapStateToProps, {
  selectProgramSetting,
  R180NGProgramSettingsRequest,
  R180NGSaveRequest,
  R180NGProgramSettingsEnrollmentRequest,
  showProgramSettingSetStudentLevelModal,
  selectSmartBarContainerDomain,
  makeSelectUserOrgUserType,
});

const withReducer = injectReducer({ key: 'r180NGProgramSettingData', reducer });
const withSaga = injectSaga({ key: 'r180NGProgramSettingData', saga });

export default compose(withRouter, withReducer, withSaga, withConnect)(R180NGSettingContainer);
