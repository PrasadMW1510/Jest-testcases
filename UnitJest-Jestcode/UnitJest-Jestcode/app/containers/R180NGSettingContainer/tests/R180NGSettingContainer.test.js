import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { shallowToJson } from 'enzyme-to-json';
import { R180NGSettingContainer } from '../R180NGSettingContainer';

describe('<R180NGSettingContainer />', () => {
  let wrapper = null;
  let mockedProgramSetting = null;
  let mockedActions = null;
  let programSettingsEnrollmentRequest = null;
  let showProgramSettingSetStudentLevelModal = null;
  let mockedSmartBarSelections = null;
  let mockIsLoading = null;
  // const mockedHandleSaveClick = null;
  describe('<R180NGSettingContainer />', () => {
    mockedProgramSetting = fromJS({
      error: false,
      loading: true,
      programSetting: { auto_level: ['1'], second_language_id: ['1'], student_level: ['1'] },
      programEnrollmentSetting: [
        {
          $: { type: 'SUBPRODUCT' },
          'application.id': ['R180NG'],
          name: 'READ 180 NG Stage B',
          students: [{ total: ['1'] }],
        },
        {
          $: { type: 'SUBPRODUCT' },
          'application.id': ['R180NG'],
          name: 'READ 180 NG Stage A Teacher',
          students: [{ total: ['1'] }],
        },
      ],
    });
    mockIsLoading = false;
    mockedActions = jest.fn();
    mockedSmartBarSelections = fromJS({ selectedCohType: 'Teacher' });
    showProgramSettingSetStudentLevelModal = jest.fn();
    programSettingsEnrollmentRequest = jest.fn();
    const mockProgramAvailable = {
      selectedProgram: {
        settings: 'Settings',
        display_name: 'READ 180 Next Generation',
        display_image: '/d4d957e1b5fed309a10581d6df938d4c.png',
      },
    };
    beforeEach(() => {
      wrapper = shallow(
        <R180NGSettingContainer
          showProgramSettingSetStudentLevelModal={showProgramSettingSetStudentLevelModal}
          rosterPage={mockProgramAvailable}
          R180NGProgramSettingsRequest={mockedActions}
          R180NGProgramSettingsEnrollmentRequest={programSettingsEnrollmentRequest}
          programSettingData={mockedProgramSetting}
          programEnrollmentCount={mockedProgramSetting}
          R180NGSaveRequest={mockedActions}
          smartBarSelections={mockedSmartBarSelections}
          enrollmentData={3}
          enrollmentCount={3}
          isLoading={mockIsLoading}
        />
      );
      wrapper.setProps({
        isLoading: true,
      });
    });
    it('Expect to render correctly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('shouldComponentUpdate', () => {
    it('to verify the component will receive props', () => {
      programSettingsEnrollmentRequest = jest.fn();
      mockedProgramSetting = fromJS({
        programSetting: [],
        loading: true,
      });
      const mockProgramAvailable = {
        selectedProgram: {
          settings: 'Settings',
          display_name: 'READ 180 Next Generation',
          display_image: '/d4d957e1b5fed309a10581d6df938d4c.png',
        },
      };
      const mockNextProgramSettingContainer = fromJS({
        error: false,
        loading: true,
        programSetting: { auto_level: ['1'], second_language_id: ['0'], student_level: ['1'] },
        programEnrollmentSetting: [
          {
            $: 'SUBPRODUCT',
            name: 'progress space',
            'application.id': 'r180NG',
            students: [{ total: ['1'] }],
          },
        ],
      });
      showProgramSettingSetStudentLevelModal = jest.fn();
      wrapper = shallow(
        <R180NGSettingContainer
          programSettingData={mockNextProgramSettingContainer}
          items={mockNextProgramSettingContainer.programEnrollmentSetting}
          enrollmentData={['1']}
          showProgramSettingSetStudentLevelModal={showProgramSettingSetStudentLevelModal}
          rosterPage={mockProgramAvailable}
          R180NGProgramSettingsRequest={mockedActions}
          R180NGProgramSettingsEnrollmentRequest={programSettingsEnrollmentRequest}
          programEnrollmentCount={mockedProgramSetting}
          R180NGSaveRequest={mockedActions}
          smartBarSelections={mockedSmartBarSelections}
          enrollmentCount={3}
          isLoading={mockIsLoading}
        />
      );
      wrapper.setProps({
        programSetting: mockNextProgramSettingContainer,
        isLoading: true,
        R180NGSaveRequest: jest.fn(),
        showProgramSettingSetStudentLevelModal: jest.fn(),
        enrollmentData: ['1'],
      });
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('verify handle student level change', () => {
      expect(wrapper.find('showProgramSettingSetStudentLevelModal').toBeCalled);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('verify handle shandleChangeOkClick', () => {
      expect(wrapper.find('handleSaveClick').toBeCalled);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('Expect to render correctly', () => {
      mockedProgramSetting = fromJS({
        error: false,
        loading: true,
        programSetting: 'admin',
        programEnrollmentSetting: 'admin',
      });
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('handleSaveClick', () => {
    it('username value has been change', () => {
      const mockedEnrollCount = '-1';
      const mockProgramAvailable = {
        selectedProgram: {
          settings: 'Settings',
          display_name: 'READ 180 Next Generation',
          display_image: '/d4d957e1b5fed309a10581d6df938d4c.png',
        },
      };
      wrapper = shallow(
        <R180NGSettingContainer
          programEnrollmentCount={mockedProgramSetting}
          showProgramSettingSetStudentLevelModal={showProgramSettingSetStudentLevelModal}
          items={mockedProgramSetting.programEnrollmentSetting}
          enrollmentData={mockedEnrollCount}
          programSettingData={mockedProgramSetting}
          rosterPage={mockProgramAvailable}
          R180NGProgramSettingsRequest={mockedActions}
          R180NGProgramSettingsEnrollmentRequest={programSettingsEnrollmentRequest}
          R180NGSaveRequest={mockedActions}
          smartBarSelections={mockedSmartBarSelections}
          enrollmentCount={3}
          isLoading={mockIsLoading}
        />
      );
      wrapper.setProps({
        showProgramSettingSetStudentLevelModal: jest.fn(),
        handleSaveClick: jest.fn(),
        R180NGSaveRequest: jest.fn(),
        programEnrollmentCount: jest.fn(),
        getEnroll: jest.fn(),
        programsSettingChanged: mockedProgramSetting,
        enrollmentData: mockedEnrollCount,
        isLoading: true,
      });
      wrapper.instance().handleSaveClick(mockedProgramSetting);
      wrapper.instance().handleSetStudentLevelClick();
      expect(shallowToJson(wrapper)).toMatchSnapshot();

      it('getEnroll to have called ', () => {
        wrapper.instance().getEnroll();
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
  });
  describe('should call Admin enrollcount', () => {
    it('to verify the component will receive props', () => {
      let mockhandleChangeClick;
      programSettingsEnrollmentRequest = jest.fn();
      mockedSmartBarSelections = fromJS({ selectedCohType: '' });
      const mockProgramSettingContainer = fromJS({
        error: false,
        loading: true,
        programSetting: 'admin',
        programEnrollmentSetting: 'admin',
      });
      const mockedEnrollCount = jest.fn();
      const mockProgramAvailable = {
        selectedProgram: {
          settings: 'Settings',
          display_name: 'READ 180 Next Generation',
          display_image: '/d4d957e1b5fed309a10581d6df938d4c.png',
        },
      };
      showProgramSettingSetStudentLevelModal = jest.fn();
      wrapper = shallow(
        <R180NGSettingContainer
          programSettingData={mockProgramSettingContainer}
          programEnrollmentCount={mockProgramSettingContainer}
          rosterPage={mockProgramAvailable}
          items={mockProgramSettingContainer.programEnrollmentSetting}
          enrollmentData={mockedEnrollCount}
          R180NGProgramSettingsRequest={mockedActions}
          R180NGProgramSettingsEnrollmentRequest={programSettingsEnrollmentRequest}
          showProgramSettingSetStudentLevelModal={showProgramSettingSetStudentLevelModal}
          handleSaveClick={mockhandleChangeClick}
          smartBarSelections={mockedSmartBarSelections}
          enrollmentCount={3}
          isLoading={mockIsLoading}
        />
      );
      wrapper.setProps({
        programSetting: mockProgramSettingContainer,
        R180NGSaveRequest: jest.fn(),
        showProgramSettingSetStudentLevelModal: jest.fn(),
        enrollmentData: mockedEnrollCount,
        loading: true,
      });
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
