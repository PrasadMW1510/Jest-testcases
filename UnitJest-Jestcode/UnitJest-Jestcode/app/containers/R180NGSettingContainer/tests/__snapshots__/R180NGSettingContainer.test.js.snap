// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<R180NGSettingContainer /> <R180NGSettingContainer /> Expect to render correctly 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": Array [
          Object {
            "$": Object {
              "type": "SUBPRODUCT",
            },
            "application.id": Array [
              "R180NG",
            ],
            "name": "READ 180 NG Stage B",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
          Object {
            "$": Object {
              "type": "SUBPRODUCT",
            },
            "application.id": Array [
              "R180NG",
            ],
            "name": "READ 180 NG Stage A Teacher",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
        ],
        "programSetting": Object {
          "auto_level": Array [
            "1",
          ],
          "second_language_id": Array [
            "1",
          ],
          "student_level": Array [
            "1",
          ],
        },
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;

exports[`<R180NGSettingContainer /> handleSaveClick username value has been change 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": "admin",
        "programSetting": "admin",
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;

exports[`<R180NGSettingContainer /> should call Admin enrollcount to verify the component will receive props 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": "admin",
        "programSetting": "admin",
      }
    }
    smartBarSelections=""
  />
</div>
`;

exports[`<R180NGSettingContainer /> shouldComponentUpdate Expect to render correctly 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": Array [
          Object {
            "$": "SUBPRODUCT",
            "application.id": "r180NG",
            "name": "progress space",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
        ],
        "programSetting": Object {
          "auto_level": Array [
            "1",
          ],
          "second_language_id": Array [
            "0",
          ],
          "student_level": Array [
            "1",
          ],
        },
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;

exports[`<R180NGSettingContainer /> shouldComponentUpdate to verify the component will receive props 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": Array [
          Object {
            "$": "SUBPRODUCT",
            "application.id": "r180NG",
            "name": "progress space",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
        ],
        "programSetting": Object {
          "auto_level": Array [
            "1",
          ],
          "second_language_id": Array [
            "0",
          ],
          "student_level": Array [
            "1",
          ],
        },
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;

exports[`<R180NGSettingContainer /> shouldComponentUpdate verify handle shandleChangeOkClick 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": Array [
          Object {
            "$": "SUBPRODUCT",
            "application.id": "r180NG",
            "name": "progress space",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
        ],
        "programSetting": Object {
          "auto_level": Array [
            "1",
          ],
          "second_language_id": Array [
            "0",
          ],
          "student_level": Array [
            "1",
          ],
        },
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;

exports[`<R180NGSettingContainer /> shouldComponentUpdate verify handle student level change 1`] = `
<div>
  <R180NGSetting
    enrollmentData={3}
    handleSaveClick={[Function]}
    isLoading={true}
    onStudentLevelClick={[Function]}
    programName={
      Object {
        "display_image": "/d4d957e1b5fed309a10581d6df938d4c.png",
        "display_name": "READ 180 Next Generation",
        "settings": "Settings",
      }
    }
    programSettingData={
      Object {
        "error": false,
        "loading": true,
        "programEnrollmentSetting": Array [
          Object {
            "$": "SUBPRODUCT",
            "application.id": "r180NG",
            "name": "progress space",
            "students": Array [
              Object {
                "total": Array [
                  "1",
                ],
              },
            ],
          },
        ],
        "programSetting": Object {
          "auto_level": Array [
            "1",
          ],
          "second_language_id": Array [
            "0",
          ],
          "student_level": Array [
            "1",
          ],
        },
      }
    }
    smartBarSelections="Teacher"
  />
</div>
`;
